src/form_beam.cpp

```diff
@@ -16,6 +16,8 @@
 
 #include "vcsbeam.h"
 
+#define NTHREADS_BEAMFORMING_KERNEL 512
```

Comment:
Is this, generally, a safe number of threads to use/assume are available?

Reply:
Yes. I could not create 1024 because there are not enough registers to instantiate so many threads in the proposed implementation.

Reply:
What I meant is whether this number could ever be too large for current or slightly older generation GPUs? If 512 threads is pretty standard since ~V100 era GPUs we'll probably be okay I think.

Reply:
Based on in-person discussions, this should not be an issue unless people are using very old hardware and/or old compilers, in which case (if it's required) the above definition can be reduced.